fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -174,6 +191,8 @@ private static Comparable toComparable(@Nullable Object obj) {
         } else if (obj instanceof UUID) {
             UUID uuid = (UUID)obj;
             return new UnsignedUUID(uuid.getMostSignificantBits(), uuid.getLeastSignificantBits());
+        } else if (obj instanceof Internal.EnumLite) {
+            return ((Internal.EnumLite)obj).getNumber();
```

Comment:
I think this fixes #763. Maybe we should include the `enumFieldsWithoutIndex` test from #770 to confirm that.